[quant][graphmode] Cleanup inplace API

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38831 [jit] Inplace construct RecursiveScriptModule
* #39122 [quant][graphmode] Rename _quantize_script.py to quantize_script.py
* **#38827 [quant][graphmode] Cleanup inplace API**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

jerryzh168 committed:
[quant][graphmode] Cleanup inplace API

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

raghuramank100 commented on torch/quantization/_quantize_script.py:
```diff
@@ -31,40 +31,40 @@ def _prepare_script(model, qconfig_dict, is_dynamic):
+    return _convert_script(model, inplace, is_dynamic=True, debug=debug)
 
-def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, is_dynamic=False, debug=False):
+def _quantize_script(model, qconfig_dict, run_fn=None, run_args=None, inplace=False, is_dynamic=False, debug=False):
```
To be consistent with eager mode, lets make all of the defaults for inplace False,just like it is done here. eager has inplace=False for prepare and convert.

jerryzh168 replied:
it makes sense for `prepare`, but I feel for `convert` we do not really need to do non inplace operation, I was initially thinking about removing the inplace option for `convert`, but later I found that we may have a use case that we prepare once and convert twice(e.g. one with debug, one with no debug) and in that case we'll need the convert to be non inplace.

jerryzh168 commented on torch/csrc/jit/passes/freeze_module.cpp:
```diff
@@ -463,7 +463,7 @@ Module freeze_module(const Module& module) {
   }
 
-  auto moduleClone = module.clone();
+  auto moduleClone = module.clone(true);
```
@bzinodev do we need to deep copy the Tensor here? I feel shallow copy is enough actually

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Cleanup inplace API"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21673481](https://our.internmc.facebook.com/intern/diff/D21673481)

[ghstack-poisoned]
